sources/packages/backend/apps/api/src/route-controllers/application-offering-change-request/_tests_/e2e/application-offering-change-request.institutions.controller.createApplicationOfferingChangeRequest.e2e-spec.ts

```diff
@@ -0,0 +1,417 @@
+    fakeOffering.studyEndDate = addDays(
+      85,
+      application.programYear.startDate,
+    ).toISOString();
+    await db.educationProgramOffering.save(fakeOffering);
+
+    const payload = {
+      applicationId: application.id,
+      offeringId: fakeOffering.id,
+      reason: "Test reason.",
+    };
+
+    let responseApplicationOfferingChangeRequest = undefined;
+    // Act/Assert
+    await request(app.getHttpServer())
+      .post(endpoint)
+      .send(payload)
+      .auth(institutionUserToken, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.CREATED)
+      .expect((response) => {
+        responseApplicationOfferingChangeRequest = response.body;
+        expect(responseApplicationOfferingChangeRequest).toHaveProperty("id");
+        expect(responseApplicationOfferingChangeRequest.id).toBeGreaterThan(0);
+      });
+
+    // Get the created application offering change request status.
+    const applicationOfferingChangeRequest =
+      await db.applicationOfferingChangeRequest.findOne({
+        select: {
+          applicationOfferingChangeRequestStatus: true,
+        },
+        where: {
+          id: responseApplicationOfferingChangeRequest.id,
+        },
+      });
+
+    // Newly created application offering change request status will be in progress with students.
+    expect(applicationOfferingChangeRequest).toHaveProperty(
+      "applicationOfferingChangeRequestStatus",
+      ApplicationOfferingChangeRequestStatus.InProgressWithStudent,
+    );
+  });
+
+  it("Should throw study overlap error when trying to submit application offering request with an offering, which overlap with students another application.", async () => {
+    // Arrange
+    const savedUser = await db.user.save(createFakeUser());
+    // Student has a completed application to the institution.
+    const existingApplicationWithOverlapStudyPeriod =
+      await saveFakeApplicationDisbursements(
+        db.dataSource,
+        { institutionLocation: collegeFLocation },
+        {
+          applicationStatus: ApplicationStatus.Completed,
+        },
+      );
+    const student = existingApplicationWithOverlapStudyPeriod.student;
+
+    // Updating study period, that belongs to the program year.
+    existingApplicationWithOverlapStudyPeriod.currentAssessment.offering.studyStartDate =
+      addDays(
+        5,
+        existingApplicationWithOverlapStudyPeriod.programYear.startDate,
+      ).toISOString();
+    existingApplicationWithOverlapStudyPeriod.currentAssessment.offering.studyEndDate =
+      addDays(
+        85,
+        existingApplicationWithOverlapStudyPeriod.programYear.startDate,
+      ).toISOString();
+
+    await db.educationProgramOffering.save(
+      existingApplicationWithOverlapStudyPeriod.currentAssessment.offering,
+    );
+
+    // Student has a new completed application to the institution, which will be requested for change.
+    const applicationRequestForChange = await saveFakeApplicationDisbursements(
```

Comment:
Dev team, I have small confusion not in the e2e test but the study period overlap validation.

Say if student has an application A and offering OA. The student has only one application.

Institution requests an application offering change for A with new offering OB which has overlapping dates with OA.

We must actually allow this change because the existing offering will be replaced and student does not have any other application. But is our current validation handling this scenario ?

Reply:
@dheepak-aot , In this TC, the student has 2 applications,
application 1,
![image](https://github.com/bcgov/SIMS/assets/77353155/6004bf5f-df11-471e-bca3-24a928b9f104)
application 2,
![image](https://github.com/bcgov/SIMS/assets/77353155/7f203b41-0520-42ae-bd3b-b01592da5c31)

And a new offering was created, that overlaps application 1,
![image](https://github.com/bcgov/SIMS/assets/77353155/a39aa84e-f717-4c09-a24d-c3106b00b0ea)

Application offering change is requested on application 2 with the new offering.

Reply:
Im good with the e2e test @ann-aot as mentioned. My question was regarding that one particular scenario while performing study period overlap validation.

Reply:
@dheepak-aot, If i got your qn, correctly, will go throw, as we do not consider the current application, 
![image](https://github.com/bcgov/SIMS/assets/77353155/7cf28952-a54d-4dd7-a5cd-1e34d22e841c)

Reply:
Ah yes, this is what I was trying to ask. Thanks for sharing.